### PR TITLE
feat(server): new storage provider

### PR DESCRIPTION
**Pull request by forehalo**

### CODE REVIEW

**Comment by forehalo** [comment]
Current dependencies on/for this PR:
* `canary`
  * **PR #5405** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5405?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #5410** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5410?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
      * **PR #5433** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5433?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #5461** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5461?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/toeverything/AFFiNE/5410?utm_source=stack-comment).

File: packages/backend/server/src/modules/storage/__tests__/fs.spec.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
+import test from 'ava';
+import { promises as fs } from 'fs';
```
Comment (Brooooooklyn) [inline packages/backend/server/src/modules/storage/__tests__/fs.spec.ts]: `node:fs`

File: packages/backend/server/src/modules/storage/__tests__/fs.spec.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+import { getStreamAsBuffer } from 'get-stream';
+import { join } from 'path';
```
Comment (Brooooooklyn) [inline packages/backend/server/src/modules/storage/__tests__/fs.spec.ts]: `node:path`

File: packages/backend/server/src/modules/storage/providers/s3.ts [hunk 3]
```diff
@@ -0,0 +1,158 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import {
+  DeleteObjectCommand,
+  GetObjectCommand,
+  ListObjectsV2Command,
+  PutObjectCommand,
+  S3Client,
+} from '@aws-sdk/client-s3';
+import { Logger } from '@nestjs/common';
+import { Readable } from 'stream';
```
Comment (Brooooooklyn) [inline packages/backend/server/src/modules/storage/providers/s3.ts]: `node:stream`

**Review by Brooooooklyn — APPROVED** [review]

**Pull request merged**